### PR TITLE
Make functions statically scoped

### PR DESCRIPTION
**Pull request by atjn**

### CODE REVIEW

**Comment by atjn** [comment]
Turns out the type checker was using statically scoped rules already 😄

File: src/main/java/dk/aau/p4/abaaja/Lib/Interpreter/Interpreter.java [hunk 1]
```diff
@@ -310,7 +312,12 @@ public Symbol resolve(FuncInvokeNode node) {
             }
             default -> {
                 FuncSymbol symbol = (FuncSymbol) resolve(node.get_id());
+
+                // Functions are statically scoped. We open the function scope in the scope it was declared in.
```
Comment (BaBrixx) [inline src/main/java/dk/aau/p4/abaaja/Lib/Interpreter/Interpreter.java:316]:
```suggestion
                // Variables are statically scoped. We open the function block in the scope it was declared in.
```

Comment (Jesper276) [reply src/main/java/dk/aau/p4/abaaja/Lib/Interpreter/Interpreter.java:316]: LGTM

Comment (NybyDK) [reply src/main/java/dk/aau/p4/abaaja/Lib/Interpreter/Interpreter.java:316]: Hello!

**Review by NybyDK — APPROVED** [review]
LGTM

**Review by TSKsmiley — APPROVED** [review]
i like :D

**Review by Jesper276 — APPROVED** [review]
LGTM vroom vroom *race car sound intensifies*

**Pull request merged**